src/Symfony/Component/Lock/Store/CombinedStore.php

```diff
@@ -26,27 +29,27 @@
  *
  * @author Jérémy Derussé <jeremy@derusse.com>
  */
-class CombinedStore implements StoreInterface, LoggerAwareInterface
+class CombinedStore implements PersistStoreInterface, BlockingStoreInterface, LoggerAwareInterface
 {
     use LoggerAwareTrait;
     use ExpiringStoreTrait;
 
-    /** @var StoreInterface[] */
+    /** @var PersistStoreInterface[] */
     private $stores;
     /** @var StrategyInterface */
     private $strategy;
 
     /**
-     * @param StoreInterface[]  $stores   The list of synchronized stores
-     * @param StrategyInterface $strategy
+     * @param PersistStoreInterface[] $stores   The list of synchronized stores
+     * @param StrategyInterface       $strategy
      *
      * @throws InvalidArgumentException
      */
     public function __construct(array $stores, StrategyInterface $strategy)
     {
         foreach ($stores as $store) {
-            if (!$store instanceof StoreInterface) {
-                throw new InvalidArgumentException(sprintf('The store must implement "%s". Got "%s".', StoreInterface::class, \get_class($store)));
+            if (!$store instanceof PersistStoreInterface) {
```

Comment:
You need to make the legacy StoreInterface extend PersistStoreInterface, otherwise this is a BC break.